From f6b905860afbc904c633f8ea3d24710be8342960 Mon Sep 17 00:00:00 2001
From: Alistair Delva <adelva@google.com>
Date: Wed, 17 Jun 2020 21:55:14 -0700
Subject: [PATCH] ANDROID: GKI: scripts: Makefile: update the lz4 command

Use 'lz4' instead of 'lz4c' and increase the compression level. This
also optimizes for decompression speed at this higher level.

[CPNOTE: 03/06/21] Lee: Asked Alistair if this makes sense to Mainline

Bug: 159285792
Bug: 160031736
Test: launch_cvd -kernel_path bzImage
Test: build/build.sh
Signed-off-by: Alistair Delva <adelva@google.com>
Change-Id: I9604cc7d1f3600dec27ceb460f7c29d977a919b1
[ebiggers: Folded in the following commit:
           bd42f569a793 ("ANDROID: GKI: scripts: Makefile: update the lz4 command (#2)")]
Signed-off-by: Eric Biggers <ebiggers@google.com>
Signed-off-by: Lee Jones <joneslee@google.com>
---
 Makefile             | 2 +-
 scripts/Makefile.lib | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 9659dc723a0a..034f3371c0b2 100644
--- a/Makefile
+++ b/Makefile
@@ -517,7 +517,7 @@ KGZIP		= gzip
 KBZIP2		= bzip2
 KLZOP		= lzop
 LZMA		= lzma
-LZ4		= lz4c
+LZ4		= lz4
 XZ		= xz
 ZSTD		= zstd
 
diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 100a386fcd71..cb9865e56a0f 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -471,10 +471,10 @@ quiet_cmd_lzo_with_size = LZO     $@
       cmd_lzo_with_size = { cat $(real-prereqs) | $(KLZOP) -9; $(size_append); } > $@
 
 quiet_cmd_lz4 = LZ4     $@
-      cmd_lz4 = cat $(real-prereqs) | $(LZ4) -l -c1 stdin stdout > $@
+      cmd_lz4 = cat $(real-prereqs) | $(LZ4) -l -12 --favor-decSpeed stdin stdout > $@
 
 quiet_cmd_lz4_with_size = LZ4     $@
-      cmd_lz4_with_size = { cat $(real-prereqs) | $(LZ4) -l -c1 stdin stdout; \
+      cmd_lz4_with_size = { cat $(real-prereqs) | $(LZ4) -l -12 --favor-decSpeed stdin stdout; \
                   $(size_append); } > $@
 
 # U-Boot mkimage
-- 
2.34.1

